src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -73,6 +79,8 @@ public ReadOnlySpan(T[] array, int start, int length)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            if (default(T) == null && array.GetType() != typeof(T[]))
```

Comment:
This check should not be in `ReadOnlySpan`. If it is in the slow Span version, it should be deleted there.